src/ServicePulse.Host/vue/src/stores/MonitoringStore.js

```diff
@@ -105,6 +103,36 @@ export const useMonitoringStore = defineStore("MonitoringStore", () => {
     endpointList.value.sort(comparator);
   }
 
+  function sortGroupedEndpointList() {
+    const sortByProperty = sortBy.value;
+    let comparator;
+    const endpointShortNameComparator = (a, b) => {
+      return isSortAscending.value ? a.shortName.localeCompare(b.shortName) : b.shortName.localeCompare(a.shortName);
+    };
+
+    if (sortByProperty === "name") {
+      comparator = (a, b) => {
+        const groupNameA = a.group;
+        const groupNameB = b.group;
+        const endpointListGroupA = a.endpoints;
+        const endpointListGroupB = b.endpoints;
+
+        // Sort each group's endpoints before sorting the group name
+        endpointListGroupA.sort(endpointShortNameComparator);
+        endpointListGroupB.sort(endpointShortNameComparator);
+
+        return isSortAscending.value ? groupNameA.localeCompare(groupNameB) : groupNameB.localeCompare(groupNameA);
+      };
+    }
+    // TODO: Determine how sorting should be handled for columns other than endpoint name
+
+    if (grouping.value.groupedEndpoints.length > 1) {
```

Comment:
since comparator isn't always set, need a check for comparator truthy here

Reply:
grouping will always be set, but comparator is only set if "sortByProperty" is equal to "name"